src/libsyncengine/update_detection/file_system_observer/fsoperationset.cpp

```diff
@@ -17,62 +17,57 @@
  */
 
 #include "fsoperationset.h"
+#include <cassert>
 
 namespace KDC {
 
 FSOperationSet::~FSOperationSet() {
     clear();
 }
 
-bool FSOperationSet::getOp(UniqueId id, FSOpPtr &opPtr, bool lockMutex /*= true*/) {
-    if (lockMutex) {
-        _mutex.lock();
-    }
+bool FSOperationSet::getOp(UniqueId id, FSOpPtr &opPtr) {
+    const std::scoped_lock lock(_mutex);
+    return getOpWithoutLock(id, opPtr);
+}
 
-    auto it = _ops.find(id);
-    if (it != _ops.end()) {
+bool FSOperationSet::getOpWithoutLock(UniqueId id, FSOpPtr &opPtr) {
+    assert(_mutex.try_lock() == false);
+    if (auto it = _ops.find(id); it != _ops.end()) {
         opPtr = it->second;
-        if (lockMutex) {
-            _mutex.unlock();
-        }
         return true;
     }
-
-    if (lockMutex) {
-        _mutex.unlock();
-    }
     return false;
 }
```

Comment:
Perhaps we should use recursive mutex instead, therefore we could keep only 1 `getOp` method.

Reply:
done 👍🏾